server/src/main/java/org/elasticsearch/client/support/AbstractClient.java

```diff
@@ -406,7 +406,7 @@ public IndexRequestBuilder prepareIndex(String index, String type) {
 
     @Override
     public IndexRequestBuilder prepareIndex(String index, String type, @Nullable String id) {
```

Comment:
This makes sense to me -- even though it's 'internal', there are many consumers of the transport API, and it could cause a lot of confusion to remove `prepareIndex('index', 'type')` and add `prepareIndex('index, id')` at the same time. 

I wonder if it'd be simplest/ most consistent to replace this with `prepareIndex('index').setId('id')` for now? Then we could add a convenience method `prepareIndex('index', 'id')` in a future major version where there isn't as much room for confusion. We can continue the discussion in your follow-up PR.